packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1664,16 +1675,21 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
 ///
 /// It is performed in a standalone widget to rebuild **only** [MediaQuery] and
 /// its dependents when `window` changes, instead of rebuilding the entire widget tree.
-class _MediaQueryFromWindow extends StatefulWidget {
-  const _MediaQueryFromWindow({Key? key, required this.child}) : super(key: key);
+class MediaQueryFromWindow extends StatefulWidget {
```

Comment:
Do we have to make this public? It probably needs better docs then.

Reply:
If we can override `MediaQuery` I think being able to reuse this functionality would be useful. 🤔

I added slightly more documentation but I am not particularly great at figuring out what is missing from the documentation so please help me by pointing out what else would be useful / could be added.

The same applies for documenting the possibility of overriding the `MediaQuery` on `WidgetsApp`.

Reply:
I think it might be better to make this a static method on MediaQuery (`MediaQuery.fromWindow`, returns an instance of the current `_MediaQueryFromWindow`). Having `MediaQueryFromWindow` at the top level feels like a weird API surface.

Reply:
@Hixie should I make `MediaQueryFromWindow` private again and just return `Widget` from the static method or should it be public?

Reply:
I left it public for now, let me know if you prefer it otherwise.

Reply:
Yes, I think it would be better private (sorry for delay answering, I was out of town). The more we make public the more API surface area we have. It just seems like it would be weird for people to see "MediaQueryFromWindow" in the list of widgets.

Reply:
(Notably, I think there's no benefit to having the widget be public, as opposed to just a static method that returns that widget.)

Reply:
No problem, thank you for the answer! 🙁 
I made it private again.